Products/GenericSetup/tool.py

```diff
@@ -921,17 +921,30 @@ def getProfileInfo(self, profile_id):
 
     @security.protected(ManagePortal)
     def getDependenciesForProfile(self, profile_id):
+
+        dependencies = None
+        profile_info = None
+
         if profile_id is None:
-            return ()
+            dependencies = ()
```

Comment:
I still prefer the early return here:  there is no point in calling `getProfileInfo` for either of these cases.